internal/cmd/createdb.go

```diff
@@ -0,0 +1,122 @@
+package cmd
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"runtime/trace"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/sqlc-dev/sqlc/internal/config"
+	"github.com/sqlc-dev/sqlc/internal/opts"
+	"github.com/sqlc-dev/sqlc/internal/quickdb"
+	pb "github.com/sqlc-dev/sqlc/internal/quickdb/v1"
+	"github.com/sqlc-dev/sqlc/internal/sql/sqlpath"
+)
+
+var createDBCmd = &cobra.Command{
+	Use:   "createdb",
+	Short: "Create an ephemeral database",
+	Args:  cobra.MinimumNArgs(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		defer trace.StartRegion(cmd.Context(), "createdb").End()
+		stderr := cmd.ErrOrStderr()
+		dir, name := getConfigPath(stderr, cmd.Flag("file"))
+		env, err := cmd.Flags().GetString("env")
+		if err != nil {
+			return err
+		}
+		code, err := CreateDB(cmd.Context(), dir, name, args, env, &Options{
+			Env:    ParseEnv(cmd),
+			Stderr: stderr,
+		})
+		if err != nil {
+			fmt.Fprintln(stderr, err.Error())
+			os.Exit(code)
+		}
+		return nil
+	},
+}
+
+func CreateDB(ctx context.Context, dir, filename string, args []string, env string, o *Options) (int, error) {
+	dbg := opts.DebugFromEnv()
+	if !dbg.ProcessPlugins {
+		return 1, fmt.Errorf("process-plugins disabled")
+	}
+	_, conf, err := o.ReadConfig(dir, filename)
+	if err != nil {
+		return 1, err
+	}
+	// Find the first SQL with a managed database
```

Comment:
I guess we don't have any obvious name we can use as an identifier, but we should probably introduce one.

Or use some combination of `engine` and `out` or whatever required parameters we have in the config?